pyblish_bumpybox/plugins/collect_existing_files.py

```diff
@@ -34,103 +31,120 @@ def version_get(self, string, prefix):
             return None
         return matches[-1:][0][1], re.search("\d+", matches[-1:][0]).group()
 
-    def scan_versions(self, instance_collection, version):
+    def get_version_collections(self, collection, version):
+        """Return all collections of previous collection versions."""
 
-        # Getting collections of all previous versions and current version
         collections = []
         for count in range(1, int(version) + 1):
-
-            # Generate collection
             version_string = "v" + str(count).zfill(len(version))
-            head = instance_collection.head.replace(
+            head = collection.head.replace(
                 "v" + version, version_string
             )
-            collection = clique.Collection(
+
+            version_collection = clique.Collection(
                 head=head.replace("\\", "/"),
-                padding=instance_collection.padding,
-                tail=instance_collection.tail
+                padding=collection.padding,
+                tail=collection.tail
             )
-            collection.version = count
+            version_collection.name = collection.name
+            version_collection.label = collection.label
+            version_collection.family = collection.family
+            version_collection.version = count
+            version_collection.nodes = collection.nodes
+            collections.append(version_collection)
 
-            collection = self.scan_collection(collection)
+        return collections
 
-            if list(collection):
-                collections.append(collection)
+    def scan_collections_files(self, collections):
+        """Return all files in the directories of the collections."""
 
-        return collections
+        scanned_dirs = []
+        files = []
+        for collection in collections:
+            scan_dir = os.path.dirname(collection.head)
+            if scan_dir not in scanned_dirs and os.path.exists(scan_dir):
+                for f in os.listdir(scan_dir):
+                    file_path = os.path.join(scan_dir, f)
+                    files.append(file_path.replace("\\", "/"))
+                scanned_dirs.append(scan_dir)
 
-    def scan_collection(self, collection):
+        return files
 
-        # Scan collection directory
-        scan_dir = os.path.dirname(collection.head)
-        if scan_dir not in self.scanned_dirs and os.path.exists(scan_dir):
-            for f in os.listdir(scan_dir):
-                file_path = os.path.join(scan_dir, f)
-                self.files.append(file_path.replace("\\", "/"))
-            self.scanned_dirs.append(scan_dir)
+    def match_files_collection(self, files, collection):
```

Comment:
Another minor thing, I'd probably expect for the collection to be the first argument in this call too.